src/diff/index.js

```diff
@@ -132,10 +132,8 @@ export function diff(parentDom, newVNode, oldVNode, context, isSvg, excessDomChi
 
 			c.base = newVNode._dom;
 
-			while (tmp=c._renderCallbacks.pop()) {
-				if (c._nextState) { c.state = c._nextState; }
-				tmp.call(c);
-			}
+			tmp = c._renderCallbacks; c._renderCallbacks=[];
+			tmp.some(cb => cb.call(c));
```

Comment:
I would prefer not returning here since this could have weird consequences if they really return a value since `.some` would bailout on first return that can be casted to `true`

Reply:
Right! Good call.